contents/stacks_and_queues/code/rust/Stack.rs

```diff
@@ -0,0 +1,45 @@
+struct Stack<T> {
+	list: Vec<T>
+}
+
+impl<T> Stack<T> {
+    fn new() -> Self {
+        Stack {
+            list: Vec::new(),
+        }
+    }
+
+    // Note that this returns a reference to the value
+    // This is in contrast to pop which gives ownership of the value
+    fn top(&self) -> Option<&T> {
+        if self.list.len() > 0 {
+            Some(&self.list[self.list.len() - 1])
```

Comment:
It might best to use the methods in `Vec` for this
```suggestion
            self.list.last()
```